src/Paramore.Brighter.Locking.DynamoDB/DynamoDbLockingProvider.cs

```diff
@@ -0,0 +1,139 @@
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE. */
+#endregion
+using Amazon.DynamoDBv2;
+using Amazon.DynamoDBv2.Model;
+
+namespace Paramore.Brighter.Locking.DynamoDb
+{
+    public class DynamoDbLockingProvider : IDistributedLock
+    {
+        private readonly IAmazonDynamoDB _dynamoDb;
+        private readonly DynamoDbLockingProviderOptions _options;
+        private readonly TimeProvider _timeProvider;
+
+        public DynamoDbLockingProvider(IAmazonDynamoDB dynamoDb, DynamoDbLockingProviderOptions options)
+            :this(dynamoDb, options, TimeProvider.System)
+        {
+        }
+
+        public DynamoDbLockingProvider(IAmazonDynamoDB dynamoDb, DynamoDbLockingProviderOptions options, TimeProvider timeProvider)
+        {
+            _dynamoDb = dynamoDb;
+            _options = options;
+            _timeProvider = timeProvider;
+        }
+
+        /// <summary>
+        /// Attempt to obtain a lock on a resource
+        /// </summary>
+        /// <param name="resource">The name of the resource to Lock</param>
+        /// <param name="cancellationToken">The Cancellation Token</param>
+        /// <returns>The id of the lock that has been acquired or null if no lock was able to be acquired</returns>
+        public async Task<string?> ObtainLockAsync(string resource, CancellationToken cancellationToken = default)
+        {
+            var lockId = Guid.NewGuid().ToString();
+
+            try
+            {
+                await _dynamoDb.PutItemAsync(BuildLockRequest(resource, lockId), cancellationToken);
+            }
+            catch (ConditionalCheckFailedException)
+            {
+                return null;
+            }
+
+            return lockId;
+        }
+
+        /// <summary>
+        /// Release a lock
+        /// </summary>
+        /// <param name="resource">The name of the resource to Lock</param>
+        /// <param name="lockId">The lock Id that was provided when the lock was obtained</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>Awaitable Task</returns>
+        public async Task ReleaseLockAsync(string resource, string lockId, CancellationToken cancellationToken = default)
+        {
+            if (_options.ManuallyReleaseLock)
+            {
+                try
+                {
+                    await _dynamoDb.DeleteItemAsync(BuildReleaseRequest(resource, lockId), cancellationToken);
+                }
+                catch (ConditionalCheckFailedException)
+                {
+                    // Log that the lease is no longer valid
```

Comment:
Generally we have a model of creating a static logger in the class, and then using that. See: https://github.com/BrighterCommand/Brighter/blob/master/src/Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs

Reply:
Ah yep, I completely missed the logging. I'll sort that out.